src/main/java/redis/clients/util/JedisClusterHashTagUtil.java

```diff
@@ -0,0 +1,41 @@
+package redis.clients.util;
+
+/**
+ * Holds various methods/utilities to manipualte and parse redis hash-tags.
+ * See <a
+ * href="http://redis.io/topics/cluster-spec">Cluster-Spec : Keys hash tags</a>
+ */
+public final class JedisClusterHashTagUtil {
+
+  private JedisClusterHashTagUtil() {
+    throw new InstantiationError("Must not instantiate this class");
+  }
+
+  public static String getHashTag(String key) {
+    return extractHashTag(key, false, false);
+  }
+
+  public static boolean isClusterCompliantMatchPattern(String matchPattern) {
+    String tag = extractHashTag(matchPattern, true, true);
+    return tag != null && !tag.isEmpty() ;
+  }
+
+  private static String extractHashTag(String key, boolean includeBrackets, boolean returnNullWhenTagNotFound) {
```

Comment:
I don't like having a function with boolean parameters (specially two) as it's super difficult to use it from somewhere else without looking the code. I'll try to think of a better alternative to this.